dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -556,8 +571,22 @@ public Boolean checkCWL(File content){
                     }
                 }
             }
-            if(versionFound && classFound){
+            if(inputFound && outputFound && classFound){
+                if(!commandFound){
+                    out("Warning: 'baseCommand' field is missing in the entry file.");
+                }
+                if(!versionFound){
+                    out("Warning: 'cwlVersion' field is missing in the entry file.");
+                }
                 return true;
+            } else if(!inputFound && !outputFound && !classFound){
+                return false;
```

Comment:
- The reason why there is no error message when all three are missing, is because even if the file is not a cwl file, it will go through both cwl and wdl validation check. If the file turns out to be not a cwl file (e.g wdl or any random file), which can be seen from all three required fields for cwl file are missing or not, it will return back "false" instead of giving out error message that it's missing the required fields for cwl and exit immediately. If it exits immediately with the error, it wont even go through the other validation check function.
- Similar reason for the two fields missing but will only return one error message. My intention was to just output each error message, so if the file is still missing the other one, the new error will be the output. But I think I will try to change it to be kept in an array or something first and output all errors in the end then call system.exit